ethers-core/src/types/transaction/eip2718.rs

```diff
@@ -204,6 +204,25 @@ impl TypedTransaction {
         self
     }
 
+    pub fn set_gas_price_for_eip_1559<T: Into<U256>>(
```

Comment:
don't like the naming because this indicates it is only  eip1559 type tx

imo it would make more sense to add a designated setter for `max_priority_fee_per_gas`